src/highlight/HighlightAnnotations.tsx

```diff
@@ -1,45 +1,170 @@
+        setMessage(text || '');
+    };
+
+    // TODO: For testing purposes only
+    handleClick = (): void => {
+        const { location, setMode, setStaged, setStatus } = this.props;
+        // Page 1 "troublesome personality" of the Van Gogh doc
+        const shapes = [
+            {
+                height: 2.0239190432,
+                width: 21.1208338959,
+                x: 38.7415889335,
+                y: 75.0991835327,
+                type: 'rect' as const,
+            },
+        ];
+
+        setMode(Mode.HIGHLIGHT);
+        setStatus(CreatorStatus.staged);
+        setStaged({ location, shapes });
+    };
+
+    handleSubmit = (): void => {
+        const { createHighlight = noop, message, staged } = this.props;
+
+        if (!staged) {
+            return;
+        }
+
+        createHighlight({ ...(staged as CreatorHighlight), message });
+    };
+
+    setHighlightRef = (highlightRef: HTMLAnchorElement): void => {
+        this.setState({ highlightRef });
+    };
+
     render(): JSX.Element {
-        const { activeAnnotationId, annotations, isCreating } = this.props;
-        const sortedAnnotations = annotations.filter(isValidHighlight).sort(sortHighlight);
+        const { activeAnnotationId, annotations, isCreating, message, staged, status } = this.props;
+        const { highlightRef } = this.state;
+        const stagedAnnotation = getStagedAnnotation(staged);
+        const canReply = status !== CreatorStatus.started && status !== CreatorStatus.init;
+        const isPending = status === CreatorStatus.pending;
 
         return (
             <>
-                {/* Layer 1: Saved annotations -- visual highlights */}
-                <HighlightCanvas activeId={activeAnnotationId} annotations={sortedAnnotations} />
+                <button
+                    onClick={this.handleClick}
+                    style={{ position: 'absolute', top: 0, left: 0, pointerEvents: 'auto' }}
+                    type="button"
+                >
+                    Set Staged
+                </button>
 
-                {/* Layer 2: Saved annotations -- interactable highlights */}
-                <HighlightList annotations={sortedAnnotations} onSelect={this.handleAnnotationActive} />
+                {/* Layer 1: Saved annotations */}
+                <HighlightList
+                    activeId={activeAnnotationId}
+                    annotations={annotations}
+                    onSelect={this.handleAnnotationActive}
+                />
 
-                {/* Layer 3: Drawn (unsaved) incomplete annotation target, if any */}
+                {/* Layer 2: Drawn (unsaved) incomplete annotation target, if any */}
                 {isCreating && <HighlightCreator className="ba-HighlightAnnotations-creator" />}
+
+                {/* Layer 3a: Staged (unsaved) highlight target, if any */}
+                {isCreating && staged && stagedAnnotation && (
+                    <SingleHighlightAnnotation
```

Comment:
This seems kind of confusing. Can we use the same components for all highlight annotations, whether there are one or multiple?